back/src/companies/sirene/trackdechets/client.ts

```diff
@@ -0,0 +1,187 @@
+  ]);
+
+  const company = {
+    siret: etablissement.siret,
+    etatAdministratif: etablissement.etatAdministratifEtablissement,
+    address: fullAddress,
+    addressVoie,
+    addressPostalCode: etablissement.codePostalEtablissement,
+    addressCity: etablissement.libelleCommuneEtablissement,
+    codeCommune: etablissement.codeCommuneEtablissement,
+    name: etablissement.denominationUniteLegale,
+    naf: etablissement.activitePrincipaleEtablissement,
+    libelleNaf: etablissement.activitePrincipaleEtablissement
+      ? libelleFromCodeNaf(etablissement.activitePrincipaleEtablissement)
+      : ""
+  };
+
+  const isEntrepreneurIndividuel =
+    etablissement.categorieJuridiqueUniteLegale === "1000";
+
+  if (isEntrepreneurIndividuel) {
+    // concatenate prénom et nom
+    company.name = [
+      etablissement.prenom1UniteLegale,
+      etablissement.nomUniteLegale
+    ]
+      .join(" ")
+      .trim();
+  }
+
+  return company;
+};
+
+/**
+ * Search a company by SIRET
+ * @param siret
+ */
+export const searchCompany = (siret: string): Promise<CompanySearchResult> =>
+  client
+    .get<GetResponse<SearchStockEtablissement>>({
+      id: siret,
+      index
+    })
+    .then(r => {
+      if (r.warnings) {
+        logger.warn(r.warnings);
+      }
+      if (r.body._source.statutDiffusionEtablissement === "N") {
+        throw new AnonymousCompanyError();
+      }
+      return searchResponseToCompany(r.body._source);
+    })
+    .catch((error: Error) => {
+      if (error instanceof AnonymousCompanyError) {
+        throw error;
+      }
+      // 404 "no results found"
+      if (error instanceof ResponseError && error.meta.statusCode === 404) {
+        throw new UserInputError(ProviderErrors.SiretNotFound, {
+          invalidArgs: ["siret"]
+        });
+      }
+      logger.error(
+        `${ProviderErrors.ServerError}: ${error.name}, ${error.message}, \n`,
+        { stacktrace: error.stack }
+      );
+      throw new Error(`${ProviderErrors.ServerError}`);
+    });
+
+/**
+ * Build a list of company objects from a full text search response
+ */
+const fullTextSearchResponseToCompanies = (
+  r: SearchHit[]
+): CompanySearchResult[] =>
+  r.map(({ _source }) => searchResponseToCompany(_source));
+
+/**
+ * Full text search
+ */
+export const searchCompanies = (
+  clue: string,
+  department?: string,
+  options?: Partial<SearchOptions>,
+  requestOptions?: Partial<TransportRequestOptions>
+): Promise<CompanySearchResult[]> => {
+  const qs = removeDiacritics(clue);
+  // Multi-match query docs https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html
+  const must: QueryDslQueryContainer[] = [
+    {
+      match: {
+        // field created during indexation from the copy of multiple fields
+        // check this in indexInseeSiret.ts and "stocketablissement" index mapping
+        td_search_companies: {
+          query: qs,
+          operator: "or"
```

Comment:
Ça aurait été chouette d'avoir une recherche fuzzy avec un Levensthein de 1 ici. J'ai essayé d'ajouter `fuzinness: 1` mais ça ne fonctionne pas. J'imagine qu'il faut surement modifier la mapping. On pourra voir ça dans un second temps.

Reply:
c'est déjà le cas par défaut:
"If not specified, the default values are 3 and 6, equivalent to AUTO:3,6" cf. https://www.elastic.co/guide/en/elasticsearch/reference/6.8/common-options.html#fuzziness